src/utils/mod.rs

```diff
@@ -38,16 +38,17 @@ impl PartialEq for Node {
 }
 
 pub fn is_a_parent_of(parent: &str, child: &str) -> bool {
-    (child.starts_with(parent) && child.chars().nth(parent.chars().count()) == Some('/'))
-        || parent == "/"
+    let path_parent = std::path::Path::new(parent);
+    let path_child = std::path::Path::new(child);
+    (path_child.starts_with(path_parent) && !path_parent.starts_with(path_child))
```

Comment:
I've been staring at this for sometime and I think it is good. 

Can you add a test for this case as it was bothering me:

`is_a_parent_of('/usr/folder', '/usr/folder_not_a_child')  == False` 

?
@rivy

Reply:
I tried to find a portable std::path-based method to determine parent-child relation and couldn't find anything. This was what I finally came up with as a possible design. But, yep, it's a bit twisty and I mentally bounced the code around for a while as well. I had to add a few test cases just to help prove it to myself. I'll happily add any extra test cases.

Reply:
Done, and it passes (unlike `std::String::starts_with()`, `std::path::Path.starts_with()` says that it "only considers whole path components to match" which is why this works).

I'll add the commit later tonight or tomorrow.